### DISCUSSION TITLE
platform.txt error on raspberry

### DISCUSSION THREAD

**Discussion by DanSkareda** [SpenceKonde/megaTinyCore#911] [Library discussion]
I have a raspberry pi 4b with newly installed raspios bullseye 32 bit. I installed the arduino IDE from the program library, it is version 1.8.13. Then I opened the "file - properties" and pasted "http://drazzy.com/package_drazzy.com_index.json". I opened the board manager and installed "megaTinyCore" version 2.6.5. Then I selected e.g. "ATtiny1614 w/optiboot" (I try also "ATTiny1614" without optiboot) and tried to compile (CTRL+R) the default empty project (I also tried a simple "blink"). The compilation ends with an error:

Warning: platform.txt from 'megaTinyCore' kernel contains deprecated compiler.path={runtime.tools.avr-gcc.path}/bin/, automatically converted to compiler.path=/usr/bin/. Consider upgrading this kernel.
In file included from /home/pi/.arduino15/packages/megaTinyCore/hardware/megaavr/2.6.5/cores/megatinycore/Arduino.h:28:0,
                 from sketch/sketch_feb11a.ino.cpp:1:
/home/pi/.arduino15/packages/megaTinyCore/hardware/megaavr/2.6.5/cores/megatinycore/core_devices.h:291:4: error: #error "No AC? No supported parts exist without one, something is wrong"
   #error "No AC? No supported parts exist without one, something is wrong"
    ^
/home/pi/.arduino15/packages/megaTinyCore/hardware/megaavr/2.6.5/cores/megatinycore/core_devices.h:300:4: error: #error "No ADC? No supported parts exist without one, something is wrong"
   #error "No ADC? No supported parts exist without one, something is wrong"
    ^
/home/pi/.arduino15/packages/megaTinyCore/hardware/megaavr/2.6.5/cores/megatinycore/core_devices.h:331:4: error: #error "No TCA? No supported parts exist without one, something is wrong"
   #error "No TCA? No supported parts exist without one, something is wrong"

...etc. there are many more errors...

If I try to compile the empty project for default "Uno", it compiles without errors. In Win11 everything described above works correctly. Please, what is wrong with Linux? Thanks.

**Comment by SpenceKonde** [comment]
GOOD GOD WHAT IS IT THIS WEEK WITH PEOPLE WITH GODAWFUL HACKED UP LINUX VERSIONS FROM PACKAGE MANAGERS?! 

Third time in the past 5 issues I've had to point this out. The app repo maintainers, who know not a thing about Arduino don't let that stop them from modifying it (most often by changign the rewrites rules and in so doing also eviscerating any third party board or core support. 

Uninstall that broken version of the IDE, downloadand install the working one from arduino.cc. and then reinstall this core through board manager and it should work. 

Never use  a version of the ide that came from a package manager. It's not the same thing.

**Comment by DanSkareda** [comment]
I did as you advise and all is well. It compiles without errors. Thank you for your advice.

**Comment by leedowthwaite** [comment]
Thanks @SpenceKonde for the clear answer here, that would have taken me a while to figure out since this is the first time I've ever done Arduino development in a Linux environment.
